src/python/gem5/resources/resource.py

```diff
@@ -554,6 +554,80 @@ def _get_weights_and_simpoints_from_file(
         return simpoint_list, weight_list
 
 
+class WorkloadResource(AbstractResource):
+    """A workload resource. This resource is used to specify a workload to run
+    on a board. It contains the function to call and the parameters to pass to
+    that function.
+    """
+
+    def __init__(
+        self,
+        function: str = None,
+        resources: Optional[Dict[str, str]] = None,
+        additional_params: Optional[Dict[str, str]] = None,
```

Comment:
I can see why you're including these, as this reflects the schema, but this is redundant with the field `parameters`. I would remove these two fields and have a parser in the constructor to create `parameters` from the JSON entry.

Reply:
I have a parser in the constructor to create `parameters`, and I have the `parameters` fields to keep backward compatibility with `CustomWorkload` as it directly passed `parameter` instead of `resources` and `additional_params`.

Reply:
I don't follow. I'm suggesting we get rid of `additional_parameters` and `resources` as parameters here. This will keep backwards compatibility with `CustomWorkload`. What we'd change is when constructing this from gem5 Resources data sources. We'd parse the `resources` and `additional_params fields into `parameters`. It shouldn't be too hard to do.

Reply:
To confirm, do you mean adding an if in `obtain_resources` for the category `workload` and doing the parsing there instead of in the constructor of the `WorkloadResource` class?